pkg/manifest/manifest_test.go

```diff
@@ -402,3 +406,32 @@ func TestSetNodeRequirements(t *testing.T) {
 		t.Errorf("expected node requirement req1 to have value val1 but was %sa", val)
 	}
 }
+
+func TestGetRequiredHooksContainsEntries(t *testing.T) {
+	builder := NewBuilder()
+	builder.SetID("reqhooks")
+	reqHooks := []string{"req-hook"}
+	config := map[interface{}]interface{}{"required_hooks": reqHooks}
+	err := builder.SetConfig(config)
+
+	Assert(t).IsNil(err, "Should not have erred setting config for test manifest")
+
+	manifest := builder.GetManifest()
+	result := manifest.GetRequiredHooks()
+
+	expectedList := testRequiredHooksList()
+	expected := expectedList[0]
+	for _, hook := range result {
+		Assert(t).AreEqual(hook, expected, "Should contain the required hook")
```

Comment:
Probably should be `for i, hook := range result {` then
` Assert(t).AreEqual(hook, expectedList[i], "Should contain the required hook")`

Reply:
Agreed that something better should be done to validate the full array. I mostly did it as a hack due to the test framework limitations. Let me see if I can improve it by using the standard test framework.

Reply:
`reflect.DeepEqual()` was the function I needed. Updated the test.